### PR TITLE
Additional arguments - alternative implementation

### PR DESCRIPTION
**Pull request by wmfw23**
An alternate implementation of argument feeding for plotting scripts (relative to [this](https://github.com/SWIFTSIM/pipeline/pull/8)), using yaml lists and cleaner parsing. Amended example and test scripts to demonstrate use.

### CODE REVIEW

**Comment by JBorrow** [comment]
This looks much better; I will get back to you on some more details and a review as soon as I can.

File: example/config/config.yml [hunk 1]
```diff
@@ -28,3 +28,6 @@ scripts:
     output_file: density_temperature.png
     section: Density-Temperature
     title: Density-Temperature
+    additional_args:
+      cmap: cividis
```
Comment (JBorrow) [inline example/config/config.yml:32]: I'd really prefer if it we can use something here that doesn't include colour maps as an example - I don't want to encourage this.

File: example/config/scripts/density_temperature.py [hunk 2]
```diff
@@ -108,8 +111,16 @@ def make_single_image(
 
     vmax = np.max([np.max(hist) for hist in hists])
 
+    cmap_suffix = ''
+    if cmap_order == 'inverted':
+        cmap_suffix = '_r'
+    elif cmap_order != 'regular':
+        raise Exception(f'colormap ordering {add_args[1]} not understood')
+
+    cmap = get_cmap(cmap+cmap_suffix)
```
Comment (JBorrow) [inline example/config/scripts/density_temperature.py:120]: File not formatted with `black`

File: example/config/scripts/density_temperature.py [hunk 3]
```diff
@@ -124,6 +135,12 @@ def make_single_image(
 
     arguments = ScriptArgumentParser(description="Basic density-temperature figure.")
 
+    # set additional argument defaults if not specified 
+    if not hasattr(arguments, "cmap_order"):
+        arguments.cmap_order = 'regular'
```
Comment (JBorrow) [inline example/config/scripts/density_temperature.py:140]: Please don't use single quotes - always double - throughout. I think `black` should take care of this?

File: swiftpipeline/argumentparser.py [hunk 4]
```diff
@@ -115,6 +119,15 @@ def __setup_parser(self):
             required=True,
         )
 
+        self.parser.add_argument(
```
Comment (JBorrow) [inline swiftpipeline/argumentparser.py:122]: My idea here was to have an API where the names were passed too. I don't like the idea of having positional-only arguments here. I'd prefer it if we could add an argument (with the long name only) and then pass that based on the input yaml. I'm happy to make this change if needed.

**Review by JBorrow — COMMENTED** [review]
I would like to have the names provided used more here, if possible. Happy to make those changes myself if you don't have time.

That would mean that each script that requires extra arguments can accept them as named, e.g.

`python3 my_script.py --cmap cividis`

File: example/config/config.yml (hunk 1 above)
Comment (wmfw23) [reply example/config/config.yml:32]: yes, no interest in changing `cmap`, just needed something to demonstrate functionality in the provided example plot. Could also have option to plot subgrid or hydro properties

File: example/config/scripts/density_temperature.py (hunk 2 above)
Comment (wmfw23) [reply example/config/scripts/density_temperature.py:120]: formatting...

File: example/config/scripts/density_temperature.py (hunk 3 above)
Comment (wmfw23) [reply example/config/scripts/density_temperature.py:140]: ok

File: swiftpipeline/argumentparser.py (hunk 4 above)
Comment (wmfw23) [reply swiftpipeline/argumentparser.py:122]: Not really sure what you mean so please feel free to add how you like after push the next commit.

**Comment by wmfw23** [comment]
Yes please feel free to change the way scripts accept arguments as I'm sure you have more experience with this - I'm really just interested in having the functionality available in some form

**Comment by JBorrow** [comment]
I'm going to merge this as-is and then open another PR with my changes. I'll also add you as a contributor.

**Pull request merged**